react-components/src/components/RevealToolbar/RuleBasedOutputsButton.tsx

```diff
@@ -43,6 +49,10 @@ export const RuleBasedOutputsButton = ({
     setRuleInstances(ruleInstancesResult.data);
   }, [ruleInstancesResult]);
 
+  useEffect(() => {
+    setCurrentRuleSetEnabled(newRuleSetEnabled);
+  }, [newRuleSetEnabled]);
```

Comment:
Do we need this? Could we just use `setCurrentRuleSetEnabled` instead of `setNewRuleSetEnabled` below?

Reply:
Nice observation. But this is to let the spinner start render one event cycle before triggering the process of setting the current rule set enabled on the next cycles.
If we set the flag IsRuleLoading on the same event cycle of setting the CurrentRuleSetEnabled, then the spinner will only shows up a bit later in the middle of the rule processing/when the event flow is free to trigger a new render.
Do you have any other suggestion to do that?